src/main/java/javax/validation/ConstraintViolationException.java

```diff
@@ -32,7 +34,7 @@ public ConstraintViolationException(String message,
 			this.constraintViolations = null;
 		}
 		else {
-			this.constraintViolations = new HashSet<ConstraintViolation<?>>( constraintViolations );
+			this.constraintViolations = new HashSet<>( constraintViolations );
```

Comment:
Thinking more about it, I don't see any real advantage of making it immutable. It's not that we'd share the set of violations between multiple invocations or threads, it's specific to one invocation of the validation engine. And maybe some user wants to amend the exception when passing it up. It seems far fetched, but I don't think we have strong reason to prevent it.